docs/proposals/04-new-core-gardener-cloud-apis.md

```diff
@@ -0,0 +1,698 @@
+    #   id: coreos_2023_4_0_64_30G_alibase_20190319.vhd
+    #
+    #
+    # Example for AWS:
+    #
+    # apiVersion: aws.provider.extensions.gardener.cloud/v1alpha1
+    # kind: ProviderProfileConfig
+    # machineImages:
+    # - name: coreos
+    #   version: 1967.5.0
+    #   regions:
+    #   - name: europe-central-1
+    #     ami: ami-0f46c2ed46d8157aa
+    #
+    #
+    # Example for Azure:
+    #
+    # apiVersion: azure.provider.extensions.gardener.cloud/v1alpha1
+    # kind: ProviderProfileConfig
+    # machineImages:
+    # - name: coreos
+    #   version: 1967.5.0
+    #   publisher: CoreOS
+    #   offer: CoreOS
+    #   sku: Stable
+    # countFaultDomains:
+    # - region: westeurope
+    #   count: 2
+    # countUpdateDomains:
+    # - region: westeurope
+    #   count: 5
+    #
+    #
+    # Example for GCP:
+    #
+    # apiVersion: gcp.provider.extensions.gardener.cloud/v1alpha1
+    # kind: ProviderProfileConfig
+    # machineImages:
+    # - name: coreos
+    #   version: 2023.5.0
+    #   image: projects/coreos-cloud/global/images/coreos-stable-2023-5-0-v20190312
+    #
+    #
+    # Example for OpenStack:
+    #
+    # apiVersion: openstack.provider.extensions.gardener.cloud/v1alpha1
+    # kind: ProviderProfileConfig
+    # machineImages:
+    # - name: coreos
+    #   version: 2023.5.0
+    #   image: coreos-2023.5.0
+    # keyStoneURL: https://url-to-keystone/v3/
+    # dnsServers:
+    # - 10.10.10.10
+    # - 10.10.10.11
+    # dhcpDomain: foo.bar
+    # requestTimeout: 30s
+    # constraints:
+    #   loadBalancerProviders:
+    #   - name: haproxy
+    #   floatingPools:
+    #   - name: fip1
+    #     loadBalancerClasses:
+    #     - name: class1
+    #       floatingSubnetID: 04eed401-f85f-4610-8041-c4835c4beea6
+    #       floatingNetworkID: 23949a30-1cdd-4732-ba47-d03ced950acc
+    #       subnetID: ac46c204-9d0d-4a4c-a90d-afefe40cfc35
+    #
+    #
+    # Example for Packet:
+    #
+    # apiVersion: packet.provider.extensions.gardener.cloud/v1alpha1
+    # kind: ProviderProfileConfig
+    # machineImages:
+    # - name: coreos
+    #   version: 2079.3.0
+    #   id: d61c3912-8422-4daf-835e-854efa0062e4
+```
+
+### `Seed` resource
+
+Special note: The proposal contains field that are not yet existing in the current `garden.sapcloud.io/v1beta1.Seed` resource, but they should be implemented (open issues that require them are linked).
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: seed-secret
+  namespace: garden
+type: Opaque
+data:
+  kubeconfig: base64(kubeconfig-for-seed-cluster)
+
+---
+apiVersion: core.gardener.cloud/v1alpha1
+kind: Seed
+metadata:
+  name: seed1
+spec:
+  provider:
+    type: <some-provider-name> # {aws,azure,gcp,...}
+    region: europe-central-1
+  # Optional list of provider profiles that a shoot may use together with this seed. An empty list means that all
+  # existing provider profiles of the same provider type are supported.
+  # profiles:
```

Comment:
Can we invert this behavior - it would be way better for `ProviderProfile` to have `seedSlector` so it can be applied to multiple Seed clusters. 

`ProviderProfile -> Shoot` and `ProviderProfile ->  Seed` is better than
`ProviderProfile -> Shoot` and `ProviderProfile <- Seed`.

Reply:
I guess that should be possible. @adracus suggested something similar a few months back.

So, let's introduce

`.spec.seedSelector` in the `ProviderProfile` that selects everything by default.